### PR TITLE
Fix templating nested vars with convert_data=False

### PR DESCRIPTION
**Pull request by mkrizek**
##### SUMMARY

Fixes https://github.com/ansible/ansible/pull/78269#issuecomment-1185370586

Regression introduced in #78259.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

`lib/ansible/template/__init__.py`

### CODE REVIEW

**Comment by mkrizek** [comment]
cc @felixfontein

**Review by felixfontein — APPROVED** [review]
Looks good to me! I also ran the terraform tests in community.general with this branch and they passed.

File: test/integration/targets/template/tasks/main.yml [hunk 1]
```diff
@@ -801,3 +801,11 @@
   vars:
     data_not_converted: "{{ lookup('template', 'json_macro.j2', convert_data=False) }}"
     data_converted: "{{ lookup('template', 'json_macro.j2') }}"
+
+- name:
```
Comment (mkrizek) [inline test/integration/targets/template/tasks/main.yml]:
```suggestion
- name: Test convert_data is correctly set to True for nested vars evaluation
```

**Comment by mkrizek** [comment]
Merging this to unbreak the `devel` branch, anything else can be done in a followup PR.

cc @sivel

**Pull request merged**